itou/templates/apply/process_details_company.html

```diff
@@ -55,34 +59,36 @@ <h3>Historique des modifications</h3>
     {% endif %}
 {% endblock %}
 
-{% block actions %}
-    {% include "apply/includes/siae_hiring_actions.html" with job_application=job_application %}
-
-    {% if job_application.to_company.is_subject_to_eligibility_rules and request.user.is_employer and job_application.state.is_accepted and job_application.approval %}
+{% block sidebar %}
+    {# Job application accepted: cancel --------------------------------------------------------------------- #}
+    {% if job_application.state.is_accepted and job_application.can_be_cancelled %}
         <div class="c-box mb-4">
-            <p>
-                <b>PASS IAE</b>
-            </p>
-            <p>
-                La consultation et la gestion du PASS IAE (exemple : « suspendre » ou « prolonger » un PASS IAE) se fait depuis le profil salarié.
-            </p>
-            <a class="btn btn-block btn-primary" href="{% url 'approvals:detail' pk=job_application.approval.id %}" {% matomo_event "salaries" "clic" "detail-salarie-depuis-candidature" %}>Accéder au profil salarié</a>
+            {% if job_application.to_company.is_subject_to_eligibility_rules %}
+                <p>
+                    <strong>Rétractation</strong>
+                </p>
+                <p>
+                    Si vous annulez cette embauche, vous ne pourrez pas prétendre à l'aide au poste pour les jours éventuellement travaillés.
+                </p>
+                <p>
+                    {{ job_application.job_seeker.get_full_name }} restera éligible à l'IAE et pourra de nouveau vous envoyer une candidature dans le futur.
+                </p>
+            {% endif %}
+            <a href="{% url 'apply:cancel' job_application_id=job_application.id %}" class="btn btn-danger btn-block btn-ico text-decoration-none">
+                <i class="ri-arrow-go-back-line font-weight-medium" aria-hidden="true"></i>
+                <span>Annuler l'embauche</span>
+            </a>
         </div>
     {% endif %}
-{% endblock %}
-
-{% block script %}
-    {{ block.super }}
-
-    <script nonce="{{ CSP_NONCE }}">
-        htmx.onLoad((target) => {
-            $(".hide-if-empty", target).each(function() {
-                if ($(this).children().length == 0) {
-                    $(this).hide();
-                } else {
-                    $(this).show();
-                }
-            })
-        })
-    </script>
+    <div class="c-box mb-4">
+        <p>
+            <b>PASS IAE</b>
+        </p>
+        <p>
+            La consultation et la gestion du PASS IAE (exemple : « suspendre » ou « prolonger » un PASS IAE) se fait depuis le profil salarié.
+        </p>
+        <a class="btn btn-block btn-primary" href="">Accéder au profil salarié</a>
```

Comment:
:eyes:

Reply:
euh, avant on avait `{% url 'approvals:detail' pk=job_application.approval.id %}" {% matomo_event "salaries" "clic" "detail-salarie-depuis-candidature" %}` et ça me semblait très bien :thinking:

Reply:
Bon là vraiment c'est nimp. Je comprends pas comment j'ai pu le supprimer sans m'en rendre compte, tester qu'il y avait bien une url sur le bouton parcque j'avais vu le href="" vide dans le code et juste en survolant le bouton me dire que c'est bon, y'a une url injecté par miracle (alors que si j'avais cliqué j'aurais vu c'etait juste l'url de la page courante) 😵‍💫

Et maintenant que j'ia remis {% url  %}, j'ai 58 tests qui plantent parcque "Failed: Undefined template variable 'job_application.approval.id'". Comment ça pouvais fonctionner avant ?

Reply:
Parce que tu as déplacé la ligne `{% if job_application.to_company.is_subject_to_eligibility_rules and request.user.is_employer and job_application.state.is_accepted and job_application.approval %}`